### PR TITLE
gui-wm/dwl: new package, migrated from ::guru, add 0.4, 9999, 9999-r1

### PR DESCRIPTION
**Pull request by sevz17**
Disclaimer: I am the maintainer of dwl.

### CODE REVIEW

File: gui-wm/dwl/dwl-9999-r1.ebuild [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+		9999)
+			EGIT_BRANCH=main
+			;;
+		9999-r1)
+			EGIT_BRANCH=wlroots-next
+			WLROOTS_SLOT="0/9999"
+			;;
+	esac
+else
+	SRC_URI="https://github.com/djpohly/${PN}/archive/v${PV}.tar.gz -> ${P}.tar.gz"
```
Comment (thesamesam) [inline gui-wm/dwl/dwl-9999-r1.ebuild:25]: If you're the maintainer, pretty please consider making real dist tarballs (possibly with the lord's xz compression)? github tarballs aren't reliable and we should avoid relying on them when we have a choice.

... although I see it's using homebrew Makefiles, not autotools or Meson, so I guess that's tricky for now.

File: gui-wm/dwl/dwl-9999-r1.ebuild [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+src_prepare() {
+	restore_config config.h
+
+	default
+}
+
+src_compile() {
+	emake PKG_CONFIG="$(tc-getPKG_CONFIG)" CC="$(tc-getCC)" \
+		XWAYLAND="$(usex X -DXWAYLAND "")" XLIBS="$(usex X "xcb xcb-icccm" "")" dwl
```
Comment (thesamesam) [inline gui-wm/dwl/dwl-9999-r1.ebuild:66]: please use usev

File: gui-wm/dwl/metadata.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE pkgmetadata SYSTEM "https://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+    <maintainer type="person" proxied="yes">
+        <email>leohdz172@proton.me</email>
+        <name>Leonardo Hernández Hernández</name>
+        <description>Upstream Maintainer (please CC on bugs)</description>
```
Comment (thesamesam) [inline gui-wm/dwl/metadata.xml:7]: you're already going to be assigned them, so just drop this bit

or change it to "Additionally the upstream maintainer"

File: gui-wm/dwl/dwl-9999-r1.ebuild (hunk 2 above)
Comment (sevz17) [reply gui-wm/dwl/dwl-9999-r1.ebuild:66]: How do I use it?

manpage says:
```
       usev <USE item>
              Like use, but also echoes USE item when use returns true.
```

Comment (sevz17) [reply gui-wm/dwl/dwl-9999-r1.ebuild]: Ah, nevermind I found it: https://mgorny.pl/articles/the-ultimate-guide-to-eapi-8.html#usev-now-accepts-a-second-argument

The manpage should be updated.

**Comment by sevz17** [comment]
Done all.

File: gui-wm/dwl/dwl-9999-r1.ebuild (hunk 2 above)
Comment (thesamesam) [reply gui-wm/dwl/dwl-9999-r1.ebuild]: Would you mind filing a bug for the man page too? I filed https://bugs.gentoo.org/914590 earlier for the devmanual.

Comment (sevz17) [reply gui-wm/dwl/dwl-9999-r1.ebuild]: [Done](https://bugs.gentoo.org/914613)

**Comment by sevz17** [comment]
Thank you!

**Comment by thesamesam** [comment]
> Thank you!

You're welcome! Consider repaying with a port to meson 👀

**Comment by sevz17** [comment]
> > Thank you!
> 
> You're welcome! Consider repaying with a port to meson 👀

Ah, unfortunately using meson doesn't adhere to project philosophy, sorry. I mean, you could say it's a single-file wayland compositor

**Comment by thesamesam** [comment]
No worries. I was sort of hoping you'd start see the pattern with all the various Prefixes fixes thus far though - it's always about the "simple" Makefiles with their own custom variable names and the rest of it. The edge cases aren't easy to handle and that's why proper build systems exist. But I won't go on anymore.

**Comment by sevz17** [comment]
Oh, yeah, I think I get it now, also after hitting two bugs I finally managed to create a prefix to test the changes.

By the way, this was a busy week, It'll be only wwo/three eapi bumps I'll do in the evening, I was thinking in send them in the pr for the next week, is it ok?

**Comment by thesamesam** [comment]
Sounds good!

**Pull request closed without merging**